### ISSUE TITLE
Vault module needs Vault base url as parameter

### ISSUE THREAD

**Issue by francisco-mata** [coder/modules#175]
Hello Coders,

Im running Coder on a K8s backend, with a NGINX Ingress, I'm trying the module of Vault, but I'm not seeing any parameter to add a Base url for my Ingress Vault, WS is constatly trying on `127.0.0.1:8200` but it should be pointing to the domain name of my Vault service. 

This is working fine:

`curl --request POST  --data '{"token": "my-token"}' https://my-vault.my-ingress.net `

Here some of the logs of the WS:

```
Archive:  vault.zip
  inflating: vault                   
Vault installed successfully!
🔑 Authenticating with Vault ...
🔑 Logging in to Vault ...
Error authenticating: Put https://127.0.0.1:8200/v1/auth/github/login: dial tcp 127.0.0.1:8200: connect: connection refused
🥳 Vault authentication complete!
You can now use Vault CLI to access secrets.
```

**Comment by matifali** [comment]
@francisco-mata Do you see a field `organization_id` in th output?
```console
Key                        Value
---                        -----
base_url                   n/a
organization               coder
organization_id            95932066
token_bound_cidrs          []
token_explicit_max_ttl     0s
token_max_ttl              0s
token_no_default_policy    false
token_num_uses             0
token_period               0s
token_policies             []
token_ttl                  0s
token_type                 default
```

**Comment by francisco-mata** [comment]
No I dont have the `organization_id` on my `vault read auth/github/config`

**Comment by francisco-mata** [comment]
Tried to overwrite it but no luck
```
vault write auth/github/config organization_id=405
Success! Data written to: auth/github/config
```

`vault read auth/github/config`

```
Key                        Value
---                        -----
base_url                   https://github.company.com/api/v3/
organization               my-org
token_bound_cidrs          []
token_explicit_max_ttl     0s
token_max_ttl              0s
token_no_default_policy    false
token_num_uses             0
token_period               0s
token_policies             [coder]
token_ttl                  0s
token_type                 default
```

Key didn't appear :( should I consider increase Vault version? Currently at v1.4.2

**Comment by matifali** [comment]
> should I consider increase Vault version? Currently at v1.4.2

I am not sure if it will solve the issue, but if you can try it, why not?

**Comment by francisco-mata** [comment]
@matifali I can confirm my vault version was too old, I now can see `organization_id`
```
/ $ vault read auth/github/config 
Key                        Value
---                        -----
base_url                   https://github.company.com/api/v3/
organization               my-org
organization_id            405
token_bound_cidrs          []
token_explicit_max_ttl     0s
token_max_ttl              0s
token_no_default_policy    false
token_num_uses             0
token_period               0s
token_policies             [coder]
token_ttl                  0s
token_type                 default
```
But Im still seeing the issue the my token created by `my-org` says my user don't belong to `my-org`

```
$ vault login -no-print -method=github -path=/github token=$(coder external-auth access-token primary-github) -address="https://my-vault.my-ingress.net/"

Error authenticating: Error making API request.

URL: PUT https://my-vault.my-ingress.net/v1/auth/github/login
Code: 500. Errors:

* user is not part of required org

```

**Comment by francisco-mata** [comment]
Eureka! I have made it :). Well I needed to add to the Github App used to create token, permissions that apply to organization.

```
Read access to members, metadata, organization administration, organization personal access token requests, and organization personal access tokens
```

**Comment by francisco-mata** [comment]
@matifali Also I found the error on the `VAULT_ADDR`. If you defined a `env {}` inside the scope of the script, it will override the module's env. I think it should expand the preovious ones and add the new ones, but i don't know if thats possible 

Something like
```
export env=$env,new_envars
```

**Comment by matifali** [comment]
Great to know that you have resolved it. Regarding the env overriding, I will take a look and see how can we make it better.

**Comment by francisco-mata** [comment]
I would be great to update doc to add more permissions to the Github app. https://coder.com/docs/v2/latest/admin/external-auth#github

![image](https://github.com/coder/modules/assets/97964900/d497747c-b155-45dc-ba49-ac79d6b99020)

Im not sure exactly, which is the one its needed to resolved properly. But I added read-only on 
```
organization administration, organization personal access token requests, and organization personal access tokens
```
So the external-auth is already ready to use Vault module, that is a most to use :smile:

**Comment by matifali** [comment]
I guess we only need organization personally access token access.

IIRC I tested this with an GitHub OAuth app instead of GitHub app so didn't faced setting permissions.

**Comment by francisco-mata** [comment]
First I had Github Oauth app for the External Auth, but in some version Coder Doc change that it was required to swap to a Github App. So we swap out.

**Comment by matifali** [comment]
Closing this as resolved and opened coder/coder#12511 for improving docs.
